lib/structures/Message.js

```diff
@@ -279,8 +288,15 @@ class Message extends Base {
         if(data.tts !== undefined) {
             this.tts = data.tts;
         }
-        if(data.attachments !== undefined) {
-            this.attachments = data.attachments;
+        if(data.attachments) {
+            for(const [id] of this.attachments.entries()) {
```

Comment:
`this.attachments.keys()` exists. Also, you could also just clear the collection, and add the new ones.

Reply:
clearing would recreate the objects, I'm doing this specifically to avoid that